src/layouts/sidebar-sidecar/server.ts

```diff
@@ -218,17 +237,39 @@ export function getStaticGenerationFunctions<
           menuItems: navDataWithFullPaths,
           // TODO: won't default after `BASE_PATHS_TO_NAMES` is replaced
           title: BASE_PATHS_TO_NAMES[basePath] || product.name,
-          overviewItemHref: `/${product.slug}/${basePath}`,
+          overviewItemHref: versionPathPart
+            ? `/${product.slug}/${basePath}/${versionPathPart}`
+            : `/${product.slug}/${basePath}`,
         },
       ]
 
+      /**
+       * Generate the arguments sent to `getDocsBreadcrumbs` based on whether or
+       * not there is a version in the current path.
+       */
+      let generatedBaseName
+      let generatedBasePath
+      let filteredPathParts
+      if (indexOfVersionPathPart >= 0) {
+        generatedBaseName = `${baseName} ${versionPathPart}`
+        generatedBasePath = `${basePath}/${versionPathPart}`
+        filteredPathParts = pathParts.filter(
+          (_, index) => index !== indexOfVersionPathPart
+        )
+      } else {
+        generatedBaseName = baseName
+        generatedBasePath = basePath
+        filteredPathParts = pathParts
+      }
```

Comment:
💭  It seems like this could all be handled in `getDocsBreadcrumbs` based on whether or not `version` is defined.

Reply:
I see what you're saying! I felt it made more sense to me to change the arguments passed in to `getDocsBreadcrumbs` than to essentially make `getDocsBreadcrumbs` modify the arguments passed to it, since that's typically dangerous to do.

Reply:
If this is a blocker for you, happy to update it!

Reply:
I think given that the version might need to be added on to different path segments, based on the convo in slack, it feels like it would be worthwhile to abstract this into `getDocsBreadcrumbs`! 😊 

Let's make the change! 💪

Reply:
Can do!